### PR TITLE
Copy edit the README.md file.

### PR DESCRIPTION
**Pull request by jpmedley**
Jeff,

Apologies. I forgot that GitHub doesn't doesn't show changes the same way as Critique. On the next one I won't change the wrapping.

Joe

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -139,18 +159,28 @@ $ sw-precache --root=dist --static-file-globs='dist/**/*.html'
 
 The `sw-precache` module exposes two methods: `generate` and `write`.
 
-#### generate(options, callback)
-`generate` takes in [options](#options), generates resulting service worker code as a string, and
-then invokes `callback(error, serviceWorkerString)`.
-In the 1.x releases of `sw-precache`, this was the default and only method exposed by the module.
+#### generate(options, callback) `generate` takes in [options](#options),
```
Comment (jeffposnick) [inline README.md]: This Markdown leads to what I assume is unintentional formatting:

image

In general, can you make sure you `View` the rendered Markdown using the GitHub web interface and check for problematic formatting? There are a few other places in this file that show similar issues (first lines of text being included in headings, for example). I can point them out one-by-one, but I think they'll stand out if you use the `View` option.

Comment (jpmedley) [inline README.md]: That was my editor's wrap command taking undue liberties. I've fixed both
generate() and write().

Joe Medley | Technical Writer, DevPlat | jmedley@google.com | 816-678-7195

Comment (jeffposnick) [inline README.md]: That doesn't resolve
```
`callback(error, serviceWorkerString)
```
which is still missing the closing ` character.

Comment (jpmedley) [inline README.md]: Fixed.

Joe Medley | Technical Writer, DevPlat | jmedley@google.com | 816-678-7195

**Comment by jeffposnick** [comment]
There are a number of places in the doc that still have incorrect Markdown rendering. I'll flag them each in this PR, but in general, paying close attention to the rendered Markdown in the GitHub web interface (e.g. at https://github.com/jpmedley/sw-precache/blob/master/README.md or via the `View` option on the file) would be greatly appreciated prior to filing a PR. That way as a reviewer, I can focus on the updated content rather than the rendering oddities.

File: README.md [hunk 2]
```diff
@@ -163,73 +196,73 @@ likely to be an issue in a production environment, it avoids cache-conflicts whe
...
 
-#### directoryIndex [String]
-Many web servers automatically treat a URL corresponding to a directory (i.e. ending in `'/'`) as
-if it were a request for a specific index file in that directory, traditionally `'index.html'`.
-`sw-precache` will take that translation into account and serve the contents a relative
-`directoryIndex` file when there's no other match for a URL ending in `'/'`.
-To turn off this behavior, set `directoryIndex` to `false` or `null`.
-To override this behavior for one or more URLs, use the `dynamicUrlToDependencies` option to
-explicitly set up mappings between a directory URL and the corresponding file to use.
+#### directoryIndex [String] Sets a default filename to return for URL's
```
Comment (jeffposnick) [inline README.md]: image

File: README.md [hunk 3]
```diff
@@ -237,30 +270,30 @@ arbitrary URL that the client generates to map to a fallback cached HTML entry.
 ideally should serve as an "application shell" that is able to load the appropriate resources
 client-side, based on the request URL.
 
-**Note**: The current implementation checks `event.request.headers.get('accept').includes('text/html')`
-to determine whether to trigger the fallback, which will be `true` for any request made for an
-HTML document, whether it's a navigation or not. Once it's more
-[widely](https://code.google.com/p/chromium/issues/detail?id=540967)
-[supported](https://bugzilla.mozilla.org/show_bug.cgi?id=1209081), the logic will be changed to
-check for `event.request.mode === 'navigate'`.
+**Note: The current implementation searches the request's `accept` header and
```
Comment (jeffposnick) [inline README.md]: image

File: README.md [hunk 4]
```diff
@@ -163,73 +196,73 @@ likely to be an issue in a production environment, it avoids cache-conflicts whe
...
 
-Default: `[/^utm_/]`
+_Default_: `[/^utm_/]`
 
-#### importScripts [Array&#x27e8;String&#x27e9;]
-If you'd like to include one or more external scripts as part of the generated service worker code,
-use this option. The scripts passed in will be passed directly to the
-[`importScripts()`](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/basic_usage#Importing_scripts_and_libraries)
-method.
+#### importScripts [Array&#x27e8;String&#x27e9;] Writes calls to
```
Comment (jeffposnick) [inline README.md]: image

File: README.md [hunk 5]
```diff
@@ -163,73 +196,73 @@ likely to be an issue in a production environment, it avoids cache-conflicts whe
...
+`home.jade`.
 
-Default: `{}`
+_Default_: `{}`
 
-#### handleFetch [boolean]
-Determines whether the `fetch` event handler is included in the generated service worker code.
-It is useful to set this to `false` in development builds, to ensure that features like live reload
-still work (otherwise, the content would always be served from the service worker cache).
+#### handleFetch [boolean] Determines whether the `fetch` event handler is
```
Comment (jeffposnick) [inline README.md]: image

File: README.md [hunk 6]
```diff
@@ -139,18 +159,31 @@ $ sw-precache --root=dist --static-file-globs='dist/**/*.html'
...
 The `sw-precache` module exposes two methods: `generate` and `write`.
 
-#### generate(options, callback)
-`generate` takes in [options](#options), generates resulting service worker code as a string, and
-then invokes `callback(error, serviceWorkerString)`.
-In the 1.x releases of `sw-precache`, this was the default and only method exposed by the module.
+#### generate(options, callback) 
+
+`generate` takes in [options](#options), generates a service worker 
+from them and passes the result to a callback fundtion, which must
```
Comment (jeffposnick) [inline README.md]: fundtion → function

File: README.md [hunk 7]
```diff
@@ -139,18 +159,31 @@ $ sw-precache --root=dist --static-file-globs='dist/**/*.html'
...
-`write` is a helper method that calls `generate` and takes the resulting string content and
-writes it to disk, at `filePath`. It then invokes `callback(error)`.
+Since 2.2.0, `generate()` also returns a
+[`Promise`](https://developer.mozilla.org/en-
+US/docs/Web/JavaScript/Reference/Global_Objects/Promise).
 
-Since 2.2.0, `write()` also returns a [`Promise`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise).
+#### write(filePath, options, callback) 
+`write` takes in [options](#options), generates a service worker from them, 
+and writes the service worker to a specified file. If any errors are
```
Comment (jeffposnick) [inline README.md]: `callback(error)` is invoked unconditionally, not just when errors are encountered.
If there are no errors, then the `error` parameter is `null`.
If there is an error, then the `error` parameter contains that error.

The is the standard Node approach to callbacks: http://thenodeway.io/posts/understanding-error-first-callbacks/

Comment (jpmedley) [inline README.md]: On Tue, Nov 10, 2015 at 2:23 PM, Jeffrey Posnick <notifications@github.com>
wrote:

> callback(error) is invoked unconditionally, not just when errors are
> encountered.
> If there are no errors, then the error parameter is null.
> If there is an error, then the error parameter contains that error.


`write` takes in [options](#options), generates a service worker from them,
and writes the service worker to a specified file. This method always
invokes `callback(error)`. If no error was found, the `error` parameter
will be `null'

Joe Medley | Technical Writer, DevPlat | jmedley@google.com | 816-678-7195

Comment (jpmedley) [inline README.md]: That's the text I added. Everything else is fixed.

Joe Medley | Technical Writer, DevPlat | jmedley@google.com | 816-678-7195

**Comment by jeffposnick** [comment]
There's one more change to make (that fundtion → function I flagged) but otherwise I think we're good. Thanks, Joe!

**Comment by jpmedley** [comment]
Fixed. The travis stuff is finishing now.

Side note: I saw '(that fundtion → function I flagged) ' and thought it was
some kind of developer shorthand. I read it as 'that function then function
or flagged' and thought WTF?

Joe Medley | Technical Writer, DevPlat | jmedley@google.com | 816-678-7195

**Pull request merged**